src/features/defi/helpers/normalizeOpportunity.tsx

```diff
@@ -114,14 +122,74 @@ const transformFoxy = (foxies: MergedFoxyOpportunity[]): EarnOpportunityType[] =
   })
 }
 
+const useTransformCosmosStaking = (
+  cosmosActiveStakingOpportunities: MergedActiveStakingOpportunity[],
+  cosmosStakingOpportunities: MergedStakingOpportunity[],
+): EarnOpportunityType[] => {
+  const translate = useTranslate()
+  if (cosmosActiveStakingOpportunities.length === 0) {
+    if (cosmosStakingOpportunities.length === 0) {
+      return []
+    }
+
+    return cosmosStakingOpportunities.map(staking => {
+      return {
+        type: DefiType.TokenStaking,
+        assetId: staking.assetId,
+        provider: chainTypeToLabel(staking.chain),
+        contractAddress: staking.address,
+        tokenAddress: staking.tokenAddress,
+        rewardAddress: '',
+        tvl: staking.tvl,
+        apy: staking.apr,
+        chain: staking.chain,
+        cryptoAmount: '',
+        fiatAmount: '',
+        showAssetSymbol: true,
+      }
+    })
+  }
+
+  return cosmosActiveStakingOpportunities
+    .map(staking => {
+      return {
+        type: DefiType.TokenStaking,
+        provider: chainTypeToLabel(staking.chain),
+        version: translate('defi.validatorMoniker', { moniker: staking.moniker }),
+        contractAddress: staking.address,
+        tokenAddress: staking.tokenAddress,
+        rewardAddress: '',
+        tvl: staking.tvl,
+        apy: staking.apr,
+        chain: staking.chain,
+        assetId: staking.assetId,
+        fiatAmount: staking.fiatAmount ?? '',
+        cryptoAmount: staking.cryptoAmount ?? '',
+        moniker: staking.moniker,
+        showAssetSymbol: true,
+      }
+    })
+    .sort((opportunityA, opportunityB) => {
+      return bnOrZero(opportunityA.cryptoAmount).gt(bnOrZero(opportunityB.cryptoAmount)) ? -1 : 1
+    })
+}
+
 type NormalizeOpportunitiesProps = {
   vaultArray: SupportedYearnVault[]
   foxyArray: MergedFoxyOpportunity[]
+  cosmosActiveStakingOpportunities: MergedActiveStakingOpportunity[]
+  cosmosStakingOpportunities: MergedStakingOpportunity[]
 }
 
 export const useNormalizeOpportunities = ({
   vaultArray,
   foxyArray,
+  cosmosActiveStakingOpportunities = [],
+  cosmosStakingOpportunities = [],
 }: NormalizeOpportunitiesProps): EarnOpportunityType[] => {
-  return [...transformFoxy(foxyArray), ...useTransformVault(vaultArray)]
+  return [
+    ...useTransformCosmosStaking(cosmosActiveStakingOpportunities, cosmosStakingOpportunities),
+    ...transformFoxy(foxyArray),
+    ...useTransformVault(vaultArray),
+  ]
```

Comment:
```suggestion
  return [
    ...transformFoxy(foxyArray),
    ...useTransformCosmosStaking(cosmosActiveStakingOpportunities, cosmosStakingOpportunities),
    ...useTransformVault(vaultArray),
  ]
```

Reply:
waiting for decision before implementing this.
Also if we go with "opportunities with coins at the top" then foxy, the logic of sorting will be more complex.

Reply:
@chxash You can look at https://github.com/shapeshift/web/pull/1304 for an idea on how to sort by one criteria, then the other, with buckets

Reply:
Perfect @reallybeard so there shouldn't be the need for additional sorting except for the order used in `useNormalizeOpportunities` in that case.

Reply:
done !